BridgeEmulator/HueEmulator3.py

```diff
@@ -518,6 +541,12 @@ def sendLightRequest(light, data):
                 elif key == "xy":
                     payload["5709"] = int(value[0] * 65535)
                     payload["5710"] = int(value[1] * 65535)
+            pprint(data)
+            if "hue" in data or "sat" in data:
+                rgbValue = hsv_to_rgb(bridge_config["lights"][light]["state"]["hue"], bridge_config["lights"][light]["state"]["sat"], bridge_config["lights"][light]["state"]["bri"]);    
+                xyValue = convert_xy(rgbValue[0], rgbValue[1], rgbValue[2])
```

Comment:
add here `pprint(rgbValue)` to validate the conversion from hsl to rgb. I believe HueMania is a hue application that send the color values in hsl format. Then monitor this output, when you set red it must output 255,0,0, when you set green it must output 0,255,0 and on blue 0,0,255

Reply:
HueManiac is exactly what I was looking for, thanks! 👍 
Gonna do some tests with it see how the conversion is working.

Reply:
hsv_to_rgb conversion seems to be going wrong

```
b'{"transitiontime":2,"on":true,"sat":229,"hue":257,"bri":4}'
DATA
{'bri': 4, 'hue': 257, 'on': True, 'sat': 229, 'transitiontime': 2}
--HSV to RGB incoming parms--
'h=46.5989010989011 s=229 v=132'
---RGB conversion result---
CASE I = 3
v=33660 t=-3100418 p=-7674480 q=-4540401
[-7674480, -4540401, 33660]
XY conversion in - out
'x=-7674480 y=-4540401 bri=33660'
'R=0 G=0 B=33660 '
{'5709': 0, '5710': 0, '5712': 2, '5851': 4}
'PAYLOAD:{"5851": 4, "5709": 0, "5710": 0, "5712": 2}'
('Test ./coap-client-linux -m put -u "Hue-Emulator-750" -k "jJ6BOoRlEVTl8Zb1" '
 '-e \'{ "3311": [{"5851": 4, "5709": 0, "5710": 0, "5712": 2}] }\' '
 '"coaps://192.168.2.4:5684/15001/65541"')

```

Reply:
Let's change the function, there are plenty examples on github
original:
```
def hsv2rgb(h, s, v):
    h = float(h)
    s = float(s)
    v = float(v)
    h60 = h / 60.0
    h60f = math.floor(h60)
    hi = int(h60f) % 6
    f = h60 - h60f
    p = v * (1 - s)
    q = v * (1 - f * s)
    t = v * (1 - (1 - f) * s)
    r, g, b = 0, 0, 0
    if hi == 0: r, g, b = v, t, p
    elif hi == 1: r, g, b = q, v, p
    elif hi == 2: r, g, b = p, v, t
    elif hi == 3: r, g, b = p, q, v
    elif hi == 4: r, g, b = t, p, v
    elif hi == 5: r, g, b = v, p, q
    r, g, b = int(r * 255), int(g * 255), int(b * 255)
    return r, g, b
```
to contain the conversion to hue 360 it must be:
`h60 = h / 10922,5`

Reply:
I think after some more tinkering I finally got it working.  
Still some division by zero in the hsv_to_rgb method and probably broke miilight by changing a conversion method, will look at that later